tcms/testcases/tests/test_views.py

```diff
@@ -228,6 +232,95 @@ def _assertTestCase(self, test_case):
         self.assertEqual(test_case.notes, self.notes)
 
 
+class TestNewCasePermission(PermissionsTestCase):
+    permission_label = 'testcases.add_testcase'
+    http_method_names = ['get', 'post']
+    url = reverse('testcases-new')
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.tester = UserFactory()
+        cls.case_status_confirmed = TestCaseStatus.objects.get(name='CONFIRMED')
+        cls.case_status_proposed = TestCaseStatus.objects.get(name='PROPOSED')
+
+        cls.product = ProductFactory()
+        cls.version = VersionFactory(value='0.1', product=cls.product)
+
+        cls.plan = TestPlanFactory(
+            author=cls.tester,
+            product=cls.product,
+            product_version=cls.version)
+
+        cls.case = TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=cls.case_status_confirmed,
+            plan=[cls.plan])
+        cls.case.save()  # will generate history object
```

Comment:
You do need some of the values for plan, category, etc but don't need to create yet another test case object in the database because it will be created when the data is POSTed successfully.

Reply:
I have removed cls.plan and have created a case. If I don't create a case object from TestCaseFactory then there won't be any entry in the database to fetch later to assert in verify_post_with_permission